sonar-plugin-api/src/main/java/org/sonar/api/resources/ResourceUtils.java

```diff
@@ -135,8 +135,9 @@ public static boolean isUnitTestFile(Resource resource) {
   }
 
   /**
-   * @return whether a resource is a library
+   * @deprecated since 5.2 No more design features
    */
+  @Deprecated
   public static boolean isLibrary(Resource resource) {
```

Comment:
:new: ![INFO](https://raw.githubusercontent.com/henryju/image-hosting/master/info.png)Do not forget to remove this deprecated code someday.[![...](https://raw.githubusercontent.com/henryju/image-hosting/master/more.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)